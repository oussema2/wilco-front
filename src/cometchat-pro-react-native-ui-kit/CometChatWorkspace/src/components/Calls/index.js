export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
