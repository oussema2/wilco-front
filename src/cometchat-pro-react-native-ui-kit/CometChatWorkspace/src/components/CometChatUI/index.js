/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
