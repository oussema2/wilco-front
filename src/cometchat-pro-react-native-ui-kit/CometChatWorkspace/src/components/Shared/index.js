export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
