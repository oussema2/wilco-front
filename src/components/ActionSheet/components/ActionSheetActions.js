import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
