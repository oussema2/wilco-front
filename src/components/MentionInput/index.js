export { default as MentionInput } from './MentionInput';
