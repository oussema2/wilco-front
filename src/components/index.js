export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
