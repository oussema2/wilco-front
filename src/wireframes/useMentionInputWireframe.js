import { useState } from 'react';
import { useRootStore } from '../providers/RootStoreProvider';
import MentionInputPresenter from '../presenters/MentionInput/MentionInputPresenter';
import Api from '../services/Api';
import EntityServiceFactory from '../services/EntityServiceFactory';
import FetchEntitiesFromRemote from '../interactors/FetchEntitiesFromRemote';
import GetEntitiesFromStore from '../interactors/GetEntitiesFromStore';

const useMentionInputWireframe = ( ) => {
	const rootStore = useRootStore();

	const createPresenter = () => {
		const {
			pilotStore, authenticationStore, hashtagStore
		} = rootStore;

		const api = new Api( { authenticationStore } );

		const hashtagService = EntityServiceFactory.forHashtags( { api } );
		const pilotService = EntityServiceFactory.forPilots( { api } );

		const fetchHashtagsFromRemote = new FetchEntitiesFromRemote( {
			store: hashtagStore,
			service: hashtagService
		} );

		const fetchPilotsFromRemote = new FetchEntitiesFromRemote( {
			store: pilotStore,
			service: pilotService
		} );

		const getHashtagsFromStore = new GetEntitiesFromStore( { store: hashtagStore } );
		const getPilotsFromStore = new GetEntitiesFromStore( { store: pilotStore } );

		return new MentionInputPresenter( {
			fetchHashtagsFromRemote,
			fetchPilotsFromRemote,
			getHashtagsFromStore,
			getPilotsFromStore
		} );
	};

	const [ presenter ] = useState( createPresenter );

	return presenter;
};

export default useMentionInputWireframe;
