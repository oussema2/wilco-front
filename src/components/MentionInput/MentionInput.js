/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import PropTypes from 'prop-types';
import { MentionInput as MentionInputNative } from 'react-native-controlled-mentions';
import { observer } from 'mobx-react-lite';
import { useFocus } from '../../hooks/useFocus';
import { useInputsStyle } from '../../hooks/useInputsStyle';
import PilotSuggestions from './components/PilotSuggestions';
import HashtagSuggestions from './components/HashtagSuggestions';
import { mentionStyles } from './MentionInput.styles';
import useMentionInputWireframe from '../../wireframes/useMentionInputWireframe';
import { useDebounce } from './components/useDebounce';
import { PART_TYPE } from '../../constants/post';

const MentionInput = ( {
	testID, value, error, disabled, onFocus, onBlur, onChange,
	placeholder, capitalizeFirstLetter, minimumLines, bold
} ) => {
	const presenter = useMentionInputWireframe();
	const [ focused, onInputFocused, onInputBlurred ] = useFocus( onFocus, onBlur );
	const [ styles, minHeight ] = useInputsStyle(
		focused, error, disabled, minimumLines, value, bold
	);

	return (
		<MentionInputNative
			testID={testID}
			textInputStyle={styles.postMessageTextInput}
			minimumLines={5}
			autoCapitalize={( capitalizeFirstLetter ) ? 'sentences' : 'none'}
			value={value}
			onFocus={onInputFocused}
			onBlur={onInputBlurred}
			onChange={onChange}
			placeholder={placeholder}
			scrollEnabled={!!presenter.hasKeyword}
			style={{
				...styles.input,
				minHeight: !presenter.hasKeyword ? minHeight : 52,
				maxHeight: presenter.hasKeyword ? 52 : undefined
			}}
			multiline
			partTypes={[
				{
					trigger: PART_TYPE.mention.trigger,
					renderSuggestions: ( { keyword, onSuggestionPress } ) => {
						useDebounce( keyword, presenter.pilotMentionInputState?.onKeywordChanged, 10 );

						return (
							<PilotSuggestions
								onSuggestionPress={onSuggestionPress}
								presenter={presenter.pilotMentionInputState}
							/>
						);
					},
					textStyle: mentionStyles.linkStyle,
					isBottomMentionSuggestionsRender: true,
					isInsertSpaceAfterMention: true,
					allowedSpacesCount: 1
				},
				{
					trigger: PART_TYPE.hashtagTrigger.trigger,
					renderSuggestions: ( { keyword, onSuggestionPress } ) => {
						useDebounce( keyword, presenter.hashtagMentionInputState?.onKeywordChanged, 10 );

						return (
							<HashtagSuggestions
								onSuggestionPress={onSuggestionPress}
								presenter={presenter.hashtagMentionInputState}
							/>
						);
					},
					textStyle: mentionStyles.linkStyle,
					isBottomMentionSuggestionsRender: true,
					isInsertSpaceAfterMention: true,
					allowedSpacesCount: 0
				},
				{
					pattern: PART_TYPE.hashtagPattern.pattern,
					textStyle: mentionStyles.linkStyle
				},
				{
					pattern: PART_TYPE.url.pattern,
					textStyle: mentionStyles.linkStyle
				},
				{
					pattern: PART_TYPE.urlShort.pattern,
					textStyle: mentionStyles.linkStyle
				},
				{
					pattern: PART_TYPE.airports.pattern,
					textStyle: mentionStyles.linkStyle
				}
			]}
		/>
	);
};

MentionInput.propTypes = {
	testID: PropTypes.string,
	value: PropTypes.string,
	error: PropTypes.string,
	disabled: PropTypes.bool,
	onFocus: PropTypes.func,
	onBlur: PropTypes.func,
	onChange: PropTypes.func,
	minimumLines: PropTypes.number,
	capitalizeFirstLetter: PropTypes.bool,
	placeholder: PropTypes.string,
	bold: PropTypes.bool
};

MentionInput.defaultProps = {
	testID: 'community-input',
	value: '',
	error: '',
	disabled: false,
	onFocus: () => {},
	onBlur: () => {},
	onChange: () => {},
	minimumLines: 1,
	capitalizeFirstLetter: false,
	placeholder: '',
	bold: true
};

export default observer( MentionInput );
