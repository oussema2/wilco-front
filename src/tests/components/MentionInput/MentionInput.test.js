import React from 'react';
import { render } from '@testing-library/react-native';
import { MentionInput } from '../../../components/MentionInput';
import * as useMentionInputWireframe from '../../../wireframes/useMentionInputWireframe';
import mockUseView from '../../mocks/mockUseView';

jest.useFakeTimers();

describe( '<MentionInput />', () => {
	let component;
	const testID = 'testing-MentionInput-Component';
	const hasKeyboard = jest.fn();

	const setUp = ( props ) => {
		mockUseView(
			useMentionInputWireframe,
			{ hasKeyboard }
		);

		component = render( <MentionInput
			testID={testID}
			{...props}
		/> );
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the MentionInput component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();
			expect( component.queryByTestId( 'pilotSuggestions-testID' ) ).toBeDefined();
			expect( component.queryByTestId( 'hashtagSuggestions-testID' ) ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with a capitalizeFirstLetter', () => {
		it( 'renders the MentionInput component correctly', () => {
			setUp( { capitalizeFirstLetter: true } );

			expect( component ).toMatchSnapshot();
		} );
	} );
} );
