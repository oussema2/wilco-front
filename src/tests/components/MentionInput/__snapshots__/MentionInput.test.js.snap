// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MentionInput /> with a capitalizeFirstLetter renders the MentionInput component correctly 1`] = `
<View>
  <TextInput
    autoCapitalize="sentences"
    minimumLines={5}
    multiline={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onSelectionChange={[Function]}
    placeholder=""
    scrollEnabled={false}
    style={
      Object {
        "backgroundColor": "#F2F2F2",
        "borderRadius": 8,
        "color": "#000000",
        "fontFamily": "NotoSansJP-Bold",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": 0,
        "lineHeight": 20,
        "maxHeight": undefined,
        "minHeight": 48,
        "paddingBottom": 16,
        "paddingHorizontal": 12,
        "paddingTop": 16,
        "textAlignVertical": "top",
      }
    }
    testID="testing-MentionInput-Component"
  >
    <Text>
      <Text>
        
      </Text>
    </Text>
  </TextInput>
  <RCTScrollView
    keyboardShouldPersistTaps="always"
    style={0}
    testID="pilotSuggestions-testID"
  >
    <View />
  </RCTScrollView>
  <RCTScrollView
    keyboardShouldPersistTaps="always"
    style={0}
    testID="hashtagSuggestions-testID"
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<MentionInput /> with the default props renders the MentionInput component correctly 1`] = `
<View>
  <TextInput
    autoCapitalize="none"
    minimumLines={5}
    multiline={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onSelectionChange={[Function]}
    placeholder=""
    scrollEnabled={false}
    style={
      Object {
        "backgroundColor": "#F2F2F2",
        "borderRadius": 8,
        "color": "#000000",
        "fontFamily": "NotoSansJP-Bold",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": 0,
        "lineHeight": 20,
        "maxHeight": undefined,
        "minHeight": 48,
        "paddingBottom": 16,
        "paddingHorizontal": 12,
        "paddingTop": 16,
        "textAlignVertical": "top",
      }
    }
    testID="testing-MentionInput-Component"
  >
    <Text>
      <Text>
        
      </Text>
    </Text>
  </TextInput>
  <RCTScrollView
    keyboardShouldPersistTaps="always"
    style={0}
    testID="pilotSuggestions-testID"
  >
    <View />
  </RCTScrollView>
  <RCTScrollView
    keyboardShouldPersistTaps="always"
    style={0}
    testID="hashtagSuggestions-testID"
  >
    <View />
  </RCTScrollView>
</View>
`;
